core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceCreateSpec.scala

```diff
@@ -0,0 +1,849 @@
+        ArgumentMatchers.any()
+      )(ArgumentMatchers.any())
+    ).thenReturn(Future(workspace))
+
+    val workspaceJobId = UUID.randomUUID()
+    val jobReport = new CreateWorkspaceV2Result()
+      .jobReport(new JobReport().id(workspaceJobId.toString).status(StatusEnum.SUCCEEDED))
+      .workspaceId(UUID.fromString(workspace.workspaceId))
+
+    when(
+      workspaceManagerDAO
+        .createWorkspaceWithSpendProfile(
+          any(),
+          ArgumentMatchers.eq(name),
+          ArgumentMatchers.eq(billingProfile.getId.toString),
+          ArgumentMatchers.eq(namespace),
+          any[Seq[String]],
+          ArgumentMatchers.eq(CloudPlatform.AZURE),
+          any[Option[WsmPolicyInputs]],
+          ArgumentMatchers.eq(testContext)
+        )
+    ).thenReturn(
+      jobReport
+    )
+    when(workspaceManagerDAO.getCreateWorkspaceResult(workspaceJobId.toString, testContext)).thenReturn(jobReport)
+    val azureStorageContainerId = UUID.randomUUID()
+    val containerResult = new CreatedControlledAzureStorageContainer().resourceId(azureStorageContainerId)
+    when(
+      workspaceManagerDAO.createAzureStorageContainer(
+        workspaceId,
+        MultiCloudWorkspaceService.getStorageContainerName(workspaceId),
+        testContext
+      )
+    )
+      .thenReturn(containerResult)
+
+    val leonardoDAO: LeonardoDAO = mock[LeonardoDAO]
+    val service = new MultiCloudWorkspaceService(
+      testContext,
+      workspaceManagerDAO,
+      mock[BillingProfileManagerDAO],
+      samDAO,
+      MultiCloudWorkspaceConfig(MultiCloudWorkspaceManagerConfig("app", 60 seconds, 120 seconds), mock[AzureConfig]),
+      leonardoDAO,
+      "MultiCloudWorkspaceService-test",
+      mock[WorkspaceManagerResourceMonitorRecordDao],
+      workspaceRepository,
+      mock[BillingRepository]
+    )
+
+    val request = WorkspaceRequest(
+      namespace,
+      name,
+      attributes
+    )
+    val result: Workspace =
+      Await.result(service.createMultiCloudWorkspaceInt(request, workspaceId, billingProfile, testContext),
+                   Duration.Inf
+      )
+
+    result.name shouldBe name
+    result.workspaceType shouldBe WorkspaceType.McWorkspace
+    result.namespace shouldEqual namespace
+
+    verify(workspaceManagerDAO).createWorkspaceWithSpendProfile(
+      ArgumentMatchers.eq(UUID.fromString(result.workspaceId)),
+      ArgumentMatchers.eq(name),
+      any(), // spend profile id
+      ArgumentMatchers.eq(namespace),
+      any[Seq[String]],
+      ArgumentMatchers.eq(CloudPlatform.AZURE),
+      any[Option[WsmPolicyInputs]],
+      ArgumentMatchers.eq(testContext)
+    )
+    verify(workspaceManagerDAO)
+      .createAzureStorageContainer(
+        UUID.fromString(result.workspaceId),
+        MultiCloudWorkspaceService.getStorageContainerName(UUID.fromString(result.workspaceId)),
+        testContext
+      )
+    verify(leonardoDAO, never()).createWDSInstance(any, any, any)
+
+  }
+
```

Comment:
What happened to "create the workspace even if WDS instance creation fails"? 

https://github.com/broadinstitute/rawls/pull/2939/files#diff-b8196c4af84188faa6eb1f65595077eb0b25f6c06f5a8e64b5c60e4832b93118L481

Reply:
That test was effectively just testing that `createWdsAppInWorkspace` doesn't throw an exception when it fails, which was already covered by a test in [`MultiCloudWorkspaceServiceCloneSpec`](https://github.com/broadinstitute/rawls/blob/03f096a99c42784f3865d15fa4ae2e0f5ac84a3d/core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceCloneSpec.scala#L639C23-L639C25).